README.md

```diff
@@ -0,0 +1,5 @@
+# Groovy-kotlin-lambda-test
+
+This repository shows how easy-to-miss bugs can be introduced with lambdas in gradle build scripts.</br>
+As gradle build scripts put all the contents of Project object into the namespace, you have to be cautious </br>
+as not to capture any part of rootProject if the lambda is supposed to be used in subprojects configuration. </br>
```

Comment:
By the way... can you see now any **automated** way of preventing such subtle issues? as it's close to impossible to spot by a human reviewer (other than just well, seeing a build error).

Do you think it's reasonable e.g. if, in git-machete-intellij-plugin, we just forbid all lambdas in `*.gradle` files via some primitive regex (like `->`)?

Reply:
Prohibit regex `->`? Sure, those functions should be in buildSrc to start with.

I suppose lamba's capture is the only place those mistakes can happen.

Reply:
Wow okay, so `arg: T -> ...` syntax does NOT allow to close over its context (aka capture surrounding vars)? 😯 so the compiler throws an error if you reference such surrounding var from an arrow-style lambda?

Reply:
Okay, so indeed sounds like no difference b/w arrow and `fun` forms, other than just ability to define the return type

Reply:
Named functions declared with _fun name(...) {...}_ can capture as well, but if you want to put a function into a val in project.extra you cannot state it's name. Only anonymous funs/lambdas are accepted.